libsolidity/inlineasm/AsmAnalysis.h

```diff
@@ -116,6 +116,13 @@ class AsmAnalyzer: public boost::static_visitor<bool>
 	EVMVersion m_evmVersion;
 	AsmFlavour m_flavour = AsmFlavour::Loose;
 	boost::optional<Error::Type> m_errorTypeForLoose;
+
+	std::map<std::string, size_t> m_references;
+	int unusedStackSlotNums = 0;
+	int reusedVariableNums = 0;
+	void decreaseReference(std::string const& name);
+
+	AsmAnalyzer& initReferencesCounter(Block const& _block);
```

Comment:
At best, there should be no changes in this file. I see that it will be difficult with the stack height checks. Perhaps we could disable most of them if we are in strict mode?